code/forms/SiteTreeURLSegmentField.php

```diff
@@ -30,7 +30,7 @@ public function getAttributes() {
 			parent::getAttributes(),
 			array(
 				'data-prefix' => $this->getURLPrefix(),
-				'data-suffix' => '?stage=Stage',
+				'data-suffix' => '?stage=' . $this->getURLSuffix(),
```

Comment:
The suffix should include the `?stage=` in it; See SiteTreeURLSegmentField::getURL(), which joins this value on as though it was a full querystring.

setURLSuffix is never set anywhere, so this change would currently break. You'll need to make sure this method defaults to `?stage=Stage` if it's not set.